File: src/main/java/VetCare/Back/model/entities/Tutor.java
package VetCare.Back.model.entities;

import VetCare.Back.model.enuns.EnumStatusTutor;
import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "tutores")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Tutor {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String nome;
    @Column(unique = true)
    private String cpf;
    private String telefone;
    @Column(unique = true)
    private String email;
    private String endereco;
    private EnumStatusTutor status = EnumStatusTutor.ATIVO;

    @OneToMany(mappedBy = "tutor", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonIgnore
    private List<Pet> pets = new ArrayList<>();
}

File: front/app/types/tutor.ts
export interface TutorRequest {
  nome: string;
  email: string;
  telefone: string;
  cpf: string;
  cep: string;
  endereco?: string;
  status?: string;
}

export interface TutorResponse {
  id: number;
  nome: string;
  email: string;
  telefone: string;
  cpf: string;
  cep: string;
  endereco: string;
  status: string;
}
